chunker/chunk_documents_excel.py

```diff
@@ -0,0 +1,304 @@
+                                               
+
+                                vertical_df = self.add_context_to_likert_scale(vertical_df, self.df.iloc[start_row, 0])
+
+
+                                chunks_to_save = self.split_large_category(vertical_df, vertical_category)
+
+                                for chunk_df, chunk_name in chunks_to_save:
+                                    #chunk_id += 1
+                                    table_title = f"{self.df.iloc[start_row, 0]}_{chunk_name}"
+                                    #logging.info(f"Iterated Chunk number {chunk_id} : {chunk_df}")
+                                    chunks.append(chunk_df)
+                                    
+                                    
+
+                            except KeyError as e:
+                                print(f"Error: {e} not found in columns.")
+                                errors.append(e)
+
+                start_row = i
+        
+
+        if start_row is not None and start_row < len(self.df) - 1:
+            for vertical_category in vertical_categories:
+                vertical_cols = self.df.columns[self.df.columns.get_level_values(0) == vertical_category]
+                if not vertical_cols.empty:
+                    vertical_cols_list = list(vertical_cols)
+                    if anchor_column not in vertical_cols_list:
+                        vertical_cols_list = [anchor_column] + vertical_cols_list
+
+                    try:
+                        vertical_df = self.df.loc[start_row+1:, vertical_cols_list]
+                        vertical_df = self.add_context_to_likert_scale(vertical_df, self.df.iloc[start_row, 0])
+
+
+                        chunks_to_save = self.split_large_category(vertical_df, vertical_category)
+                        
+
+                        for chunk_df, chunk_name in chunks_to_save:
+                            table_title = f"{self.df.iloc[start_row, 0]}_{chunk_name}"
+                            #chunk_id += 1
+                            #Uncomment this if you want to see the Iterated Chunks
+                            #logging.info(f"Iterated Chunk number {chunk_id} : {chunk_df}")
+                            chunks.append(chunk_df)
+                            
+                            
+
+                            
+                         
+                    except KeyError as e:
+                        print(f"Error: {e} not found in columns.")
+                        errors.append(e)
+
+        return chunks, errors
+                        
+def get_chunk(content, url):
+
+    chunk = {
+        "url": url,
+        "filepath": get_filename(url),
+        "content": content,
+        
+    }
+    return chunk
+
+def chunk_document(data):
+
+    #For testing using Localtest container, change the file_utils.py path 
+    filepath = f"{data['documentUrl']}{data['documentSasToken']}"
```

Comment:
maybe worth double checking here. A "?" may be required between the url and the SAS token